rubicon_ml/client/mixin.py

```diff
@@ -189,6 +190,51 @@ def log_conda_environment(self, artifact_name: Optional[str] = None) -> Artifact
 
         return artifact
 
+    @failsafe
+    def log_h2o_model(
+        self,
+        h2o_model,
+        artifact_name: Optional[str] = None,
+        export_cross_validation_predictions: bool = False,
+        **log_artifact_kwargs,
+    ) -> Artifact:
+        """Log an `h2o` model as an artifact using `h2o.save_model`.
+
+        Parameters
+        ----------
+        h2o_model : h2o.model.ModelBase
+            The `h2o` model to log as an artifact.
+        artifact_name : str, optional (default None)
+            The name of the artifact. Defaults to None, using `h2o_model`'s class name.
+        export_cross_validation_predictions: bool, optional (default False)
+            Passed directly to `h2o.save_model`.
+        log_artifact_kwargs : dict
+            Additional kwargs to be passed directly to `self.log_artifact`.
+        """
+        import h2o
+
+        if self.repository.PROTOCOL == "memory":
+            raise RubiconException("`h2o` models cannot be logged in memory with `log_h2o_model`.")
+
+        if artifact_name is None:
+            artifact_name = h2o_model.__class__.__name__
+
+        with tempfile.TemporaryDirectory() as temp_dir_name:
+            model_data_path = h2o.save_model(
+                h2o_model,
+                export_cross_validation_predictions=export_cross_validation_predictions,
+                filename=artifact_name,
+                path=temp_dir_name,
+            )
+
+            artifact = self.log_artifact(
+                name=artifact_name,
+                data_path=model_data_path,
+                **log_artifact_kwargs,
+            )
```

Comment:
this pattern is probably reusable. xgboost's model saving function immediately comes to mind. we should look into expanding and generalizing this